src/types/opt.rs

```diff
@@ -0,0 +1,95 @@
+use std::fmt;
+
+use crate::NativeRepr;
+
+/// A convenience type to explicitly mark
+/// a function argument as `opt T`.
+///
+/// When left unspecified on Redscript side,
+/// it translates to its `Default` representation.
+#[derive(Default, Debug, Clone, PartialEq, PartialOrd)]
+pub enum Opt<T: NativeRepr> {
```

Comment:
```suggestion
pub enum Opt<T: NativeRepr> {
```
I think the `NativeRepr` constraint should also be removed from here as mentioned in the previous conversation, the general guidelines is they should be avoided unless they're specifically needed for an associated type or Drop and it's not needed here

Reply:
I agree, but at the same time `Opt` only exists as a convenience for `NativeRepr` types : I can remove it but it's of no use if `T` is not `NativeRepr` in the first place.

Reply:
it's the same for a lot of other things like `RedArray`, but I don't think they should all be constrained, because that forces repetetive bounds everywhere

Reply:
ok done in 82796a9 👌